### PR TITLE
feat: Implement more functions for `pixi add --pypi`

### PR DESCRIPTION
**Pull request by wolfv**
This pr adds:
```
pixi add --pypi "boltons @ https://files.pythonhosted.org/packages/46/35/e50d4a115f93e2a3fbf52438435bb2efcf14c11d4fcd6bdcd77a6fc399c9/boltons-24.0.0-py3-none-any.whl"
pixi add --pypi "exchangelib @ git+https://github.com/ecederstrand/exchangelib"
pixi add --pypi "project @ file:///absolute/path/to/project"
pixi add --pypi --editable "project @ file:///absolute/path/to/project"
```

TODO:
- [ ] Edit: I don't see a way to do this.... Improve UX of the `path` style. (now seems to need absolute path, but this is not pixi like)

### CODE REVIEW

**Comment by ruben-arts** [comment]
Did you try this? As the problem when I tried this was in the parsing of the url itself.

**Comment by wolfv** [comment]
thanks for taking this over! :)

File: src/cli/add.rs [hunk 1]
```diff
@@ -163,13 +169,28 @@ pub async fn execute(args: Args) -> miette::Result<()> {
...
+            //         let name = PyPiPackageName::from_normalized(req.name.clone());
+            //         let mut requirement = PyPiRequirement::from(req);
+            //         if args.editable {
+            //             requirement.set_editable(true);
+            //         }
+            //         Ok((name, requirement))
+            //     })
+            //     .collect::<Result<Vec<_>, uv_normalize::InvalidNameError>>()
+            //     .into_diagnostic()?;
+
```
Comment (tdejager) [inline src/cli/add.rs:185]: Remove?

File: src/project/manifest/python.rs [hunk 2]
```diff
@@ -156,6 +156,18 @@ impl PyPiRequirement {
...
     }
+
+    /// Define whether the requirement is editable.
+    pub fn set_editable(&mut self, editable: bool) {
+        match self {
+            PyPiRequirement::Path { editable: e, .. } => {
+                *e = Some(editable);
+            }
+            _ => {
+                tracing::warn!("Trying to set editable on a non-path requirement");
```
Comment (tdejager) [inline src/project/manifest/python.rs:167]: This can be an error

Comment (ruben-arts) [reply src/project/manifest/python.rs]: I made it an non blocking error

**Review by tdejager — APPROVED** [review]

**Pull request merged**